src/fluree/db/api.cljc

```diff
@@ -635,6 +636,216 @@
+        :auto - Fast-forward when possible
+        :only - Only allow fast-forward (fail otherwise)
+        :never - Never fast-forward, always replay
+      :squash? - Combine all commits into one (default false)
+      :atomic? - All-or-nothing vs apply-until-conflict (default true)
+      :selector - Which commits to include (default nil = all after LCA)
+        nil - All commits after LCA
+        {:t {:from 42 :to 44}} - Specific t-value range
+        {:t {:from 42 :until :conflict}} - Until conflict
+        {:shas ['sha1' 'sha2']} - Specific commits
+      :preview? - Dry run without making changes (default false)
+      
+  Returns promise resolving to:
+    {:status :success|:conflict|:error
+     :operation :rebase
+     :from '...' :to '...'
+     :strategy 'fast-forward'|'squash'|'replay'
+     :commits {:applied [...] :skipped [...] :conflicts [...]}
+     :new-commit 'sha'}
+     
+  Phase 1 implements: fast-forward and squash modes.
+  Phase 2 will add: cherry-pick and non-atomic modes."
+  ([conn from to]
+   (rebase! conn from to {}))
+  ([conn from to opts]
+   (validate-connection conn)
+   (promise-wrap
+    (merge/rebase! conn from to opts))))
+
+(defn reset-branch!
+  "Resets a branch to a previous state.
+  
+  Two modes available:
+  - Safe mode (default): Creates new commit reverting to target state
+  - Hard mode: Moves branch pointer (rewrites history)
+  
+  Parameters:
+    conn - Connection object
+    branch - Target branch to reset (e.g., 'ledger:main')
+    to - Target state:
+      {:t 90} - Reset to transaction t-value
+      {:sha 'abc123'} - Reset to specific commit SHA
+    opts - Map with optional reset options:
+      :mode - Reset mode (default :safe)
+        :safe - Create revert commit (non-destructive)
+        :hard - Move branch pointer (destructive)
+      :archive - How to archive on hard reset (default {:as :tag})
+        {:as :tag :name 'backup'} - Create tag at old HEAD
+        {:as :branch :name 'backup'} - Create branch at old HEAD
+        {:as :none} - No archive (requires force?)
+      :force? - Required for hard reset without archive (default false)
+      :message - Commit message for safe mode (auto-generated if not provided)
+      :preview? - Dry run without making changes (default false)
+      
+  Returns promise resolving to:
+    {:status :success|:error
+     :operation :reset
+     :branch '...'
+     :mode :safe|:hard
+     :reset-to {:t 90}|{:sha '...'}
+     :new-commit 'sha'  ; For safe mode
+     :archived {:type :tag :name '...'}  ; For hard mode
+     :previous-head 'sha'}
+     
+  Phase 1 implements: safe reset only.
+  Phase 2 will add: hard reset with archiving."
+  ([conn branch to]
+   (reset-branch! conn branch to {}))
+  ([conn branch to opts]
+   (validate-connection conn)
+   (promise-wrap
+    (merge/reset-branch! conn branch to opts))))
+
+;; Legacy API - Deprecated, use rebase! instead
+(defn merge-branches!
```

Comment:
This should be marked with deprecated metadata.

Reply:
Addressed in e84ccafa84a4d19aaeec540ce075907129e9fbbf